### PR TITLE
MAINT: Expand lint for *.py

### PR DESCRIPTION
**Pull request by gfyoung**
Expand lint to include all Python files, excluding the directories `pandas/rpy` (deprecated and soon to be removed) and `pandas/src` (C code).

### CODE REVIEW

**Comment by jreback** [comment]
are flake8 options not strict enough? IOW why were these not caught before?

**Comment by gfyoung** [comment]
@jreback : TBH, I have no idea.  I didn't really change anything with the command flags.  But if it works, that's work counts I guess. :smile:

**Comment by gfyoung** [comment]
@jreback : Is this good to merge since Travis is passing?

File: pandas/io/parsers.py [hunk 1]
```diff
@@ -2194,16 +2194,17 @@ def _handle_usecols(self, columns, usecols_key):
         usecols_key is used if there are string usecols.
         """
         if self.usecols is not None:
-            if any([isinstance(u, string_types) for u in self.usecols]):
+            if any([isinstance(usecol, string_types)
+                    for usecol in self.usecols]):
```
Comment (jorisvandenbossche) [inline pandas/io/parsers.py]: is this needed to pass the tests? (I can't imagine we nowhere use `i` in a iteration) as I actually find the one-liner better

Comment (gfyoung) [inline pandas/io/parsers.py]: @jorisvandenbossche : The `u` shadows the `u` import from `pandas.compat`.  I don't see any problem with making this expanded as `usecol`.

Comment (jorisvandenbossche) [inline pandas/io/parsers.py]: ah ok, could also use `c` or `col` 

It's not a 'problem', but IMO a multiline list comprehension or multiline if condition is always less clear than a single line one (if you are not too brief to get it on one line).

Comment (gfyoung) [inline pandas/io/parsers.py]: That works.

**Comment by gfyoung** [comment]
@jorisvandenbossche : Any reason why you moved it to 0.20.0?  This isn't an API change.

**Comment by jorisvandenbossche** [comment]
It doesn't matter to merge, that just indicates that it is not essential to backport

**Comment by gfyoung** [comment]
Ah, okay.  Good to know!

**Comment by jorisvandenbossche** [comment]
@gfyoung thanks!

**Pull request merged**